### PR TITLE
Add request logging

### PR DESCRIPTION
**Pull request by Maan2003**
use `RUST_LOG="fedimint_server::request,fedimint_client::request"` to log requests and corresponding responses.

It adds `+ Debug` at some places, should be fine IMO.

### CODE REVIEW

**Comment by dpc** [comment]
Good thinking, but it will conflict with #1323 where I'm already adding some of it. Please not land, and take a look at #1323 - possibly we can improve logging there even further.

**Comment by Maan2003** [comment]
lets drop client side logging for now then.

**Review by dpc — APPROVED** [review]

**Comment by dpc** [comment]
@Maan2003 The https://github.com/fedimint/fedimint/pull/1323 landed, and I see that you're way is very nicely done, so you might want to do it the same way on the client now.

**Comment by dpc** [comment]
@Maan2003 

This (I think) started printing every "not found" (which is an expected return in many cases) as error log level every time.

```
mint 1: 2023-01-18T06:28:46.820367Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 0: 2023-01-18T06:28:46.820415Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 3: 2023-01-18T06:28:46.820865Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 2: 2023-01-18T06:28:46.822909Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 0: 2023-01-18T06:28:46.843953Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 1: 2023-01-18T06:28:46.843881Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }   
mint 3: 2023-01-18T06:28:46.843963Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 2: 2023-01-18T06:28:46.845076Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 0: 2023-01-18T06:28:46.887150Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }   
mint 2: 2023-01-18T06:28:46.887190Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 1: 2023-01-18T06:28:46.887235Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 3: 2023-01-18T06:28:46.887222Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }   
mint 2: 2023-01-18T06:28:46.970425Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 0: 2023-01-18T06:28:46.970546Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 1: 2023-01-18T06:28:46.970627Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }   
mint 3: 2023-01-18T06:28:46.970672Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 3: 2023-01-18T06:28:47.134475Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 2: 2023-01-18T06:28:47.134475Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 0: 2023-01-18T06:28:47.134475Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 1: 2023-01-18T06:28:47.134475Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 0: 2023-01-18T06:28:47.457253Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }   
mint 1: 2023-01-18T06:28:47.457253Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 2: 2023-01-18T06:28:47.457253Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }                                                                            
mint 3: 2023-01-18T06:28:47.457279Z ERROR fedimint_server::request: error=ApiError { code: 404, message: "transaction not found" }   
```

I think it is the `#[instrument]` macro, but I don't see why.

**Comment by Maan2003** [comment]
Ah, `err` in instrument just always use `ERROR` level

**Comment by elsirion** [comment]
Are you on it @Maan2003 or should I make it an issue?

File: fedimint-api/src/module/mod.rs [hunk 1]
```diff
@@ -159,15 +160,41 @@ pub struct ApiEndpoint<M> {
...
+            state: &'a E::State,
+            dbtx: fedimint_api::db::DatabaseTransaction<'b>,
+            param: E::Param,
+        ) -> Result<E::Response, ApiError>
+        where
+            E: TypedApiEndpoint,
+            E::Param: Debug,
+            E::Response: Debug,
+        {
+            tracing::trace!(target: "fedimint_server::request", ?param, "recieved request");
```
Comment (dpc) [inline fedimint-api/src/module/mod.rs:186]: This is too low, IMO.

Comment (Maan2003) [reply fedimint-api/src/module/mod.rs:186]: :shrug: I just specify `RUST_LOG="fedimint_server::request,...."` when I want to see the requests.

Comment (justinmoon) [reply fedimint-api/src/module/mod.rs:186]: @Maan2003 Can you PR a `docs/debugging.md` file with just this suggestion. We should try to start compiling some of these debugging tips and tricks.

Comment (dpc) [reply fedimint-api/src/module/mod.rs:186]: BTW. I also think `fedimint_server` here is not needed. We should just make it on of the categories. See #1469.

**Pull request merged**